### PR TITLE
Code Quality Python Tests

### PR DESCRIPTION
**Pull request by maximiliankaul**
Simple PR changing all
- `.first()` calls to `.firstOrNull()`
- `as?` with `assertIs` #1737 
- `.refersTo` with `assertRefersTo` #1738 
- `.invokes` with `assertInvokes`
- and applying all editor code suggestions.

### CODE REVIEW

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 1]
```diff
@@ -104,7 +104,7 @@ class PythonFrontendTest : BaseTest() {
         val p = tu.namespaces["function"]
         assertNotNull(p)
 
-        val foo = p.declarations.first() as? FunctionDeclaration
+        val foo = p.declarations.firstOrNull() as? FunctionDeclaration
         assertNotNull(foo)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:108]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 2]
```diff
@@ -137,7 +137,7 @@ class PythonFrontendTest : BaseTest() {
 
         assertFullName("print", callExpression)
 
-        val literal = callExpression.arguments.first() as? Literal<*>
+        val literal = callExpression.arguments.firstOrNull() as? Literal<*>
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:140]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 3]
```diff
@@ -200,7 +200,7 @@ class PythonFrontendTest : BaseTest() {
         val body = main.body as? Block
         assertNotNull(body)
 
-        val sel = (body.statements.first() as? AssignExpression)?.declarations?.first()
+        val sel = (body.statements.firstOrNull() as? AssignExpression)?.declarations?.firstOrNull()
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:203]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 4]
```diff
@@ -281,10 +281,10 @@ class PythonFrontendTest : BaseTest() {
         val main = p.functions["foo"]
         assertNotNull(main)
 
-        val assignExpr = (main.body as? Block)?.statements?.first() as? AssignExpression
+        val assignExpr = (main.body as? Block)?.statements?.firstOrNull() as? AssignExpression
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:284]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 5]
```diff
@@ -419,8 +419,9 @@ class PythonFrontendTest : BaseTest() {
 
         // self.somevar = i
         val someVarDeclaration =
-            ((bar.body as? Block)?.statements?.get(0) as? AssignExpression)?.declarations?.first()
-                as? FieldDeclaration
+            ((bar.body as? Block)?.statements?.get(0) as? AssignExpression)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:422]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 6]
```diff
@@ -567,13 +568,16 @@ class PythonFrontendTest : BaseTest() {
 
         val ifThen = (countStmt.thenStatement as? Block)?.statements?.get(0) as? CallExpression
         assertNotNull(ifThen)
-        assertEquals(methCount, ifThen.invokes.first())
-        assertEquals(countParam, (ifThen.arguments.first() as? Reference)?.refersTo)
+        assertEquals(methCount, ifThen.invokes.firstOrNull())
+        assertEquals(countParam, (ifThen.arguments.firstOrNull() as? Reference)?.refersTo)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:572]: assertRefersTo

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 7]
```diff
@@ -784,7 +788,7 @@ class PythonFrontendTest : BaseTest() {
         val main = p.functions["main"]
         assertNotNull(main)
 
-        val mainBody = (main as? FunctionDeclaration)?.body as? Block
+        val mainBody = main.body as? Block
         assertNotNull(mainBody)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:792]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/statementHandler/StatementHandlerTest.kt [hunk 8]
```diff
@@ -202,13 +202,13 @@ class StatementHandlerTest : BaseTest() {
         // Test for `del my_list[2]`
         val deleteStmt2 = deleteExpressions[1]
         assertEquals(1, deleteStmt2.operands.size)
-        assertTrue(deleteStmt2.operands.first() is SubscriptExpression)
+        assertTrue(deleteStmt2.operands.firstOrNull() is SubscriptExpression)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/statementHandler/StatementHandlerTest.kt:205]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/statementHandler/StatementHandlerTest.kt [hunk 9]
```diff
@@ -202,13 +202,13 @@ class StatementHandlerTest : BaseTest() {
...
         assertTrue(deleteStmt2.additionalProblems.isEmpty())
 
         // Test for `del my_dict['b']`
         val deleteStmt3 = deleteExpressions[2]
         assertEquals(1, deleteStmt3.operands.size)
-        assertTrue(deleteStmt3.operands.first() is SubscriptExpression)
+        assertTrue(deleteStmt3.operands.firstOrNull() is SubscriptExpression)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/statementHandler/StatementHandlerTest.kt:211]: assertIs

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 10]
```diff
@@ -253,17 +253,17 @@ class PythonFrontendTest : BaseTest() {
...
+        val c1 = s1.declarations.firstOrNull()
         assertNotNull(c1)
         assertLocalName("c1", c1)
         val ctor = c1.firstAssignment as? ConstructExpression
         assertNotNull(ctor)
-        assertEquals(ctor.constructor, cls.constructors.first() as? ConstructorDeclaration)
+        assertEquals(ctor.constructor, cls.constructors.firstOrNull())
         assertFullName("simple_class.SomeClass", c1.type)
 
         assertEquals(c1, (s2.base as? Reference)?.refersTo)
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:264]: assertRefersTo

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt [hunk 11]
```diff
@@ -115,13 +115,13 @@ class PythonFrontendTest : BaseTest() {
         assertNotNull(callExpression)
 
         assertLocalName("bar", callExpression)
-        assertEquals(bar, callExpression.invokes.first())
+        assertEquals(bar, callExpression.invokes.firstOrNull())
```
Comment (oxisto) [inline cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:118]: assertInvokes

File: cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt (hunk 1 above)
Comment (maximiliankaul) [reply cpg-language-python/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/python/PythonFrontendTest.kt:108]: I wasn't planning on having all this improvements here (and opened separate issues), but it makes sense to have all those minor improvements in one PR -> WIP

**Review by oxisto — APPROVED** [review]

**Pull request merged**